examples/freertos_plus_tcp/z_sub_st.c

```diff
@@ -27,13 +27,17 @@
 #endif
 
 #define KEYEXPR "demo/example/**"
+#define N 10
```

Comment:
We should align the behaviour with all other bindings, I believe we don't wait for just `10` samples by default.

Reply:
Other bindings do not implement single-threaded pub/sub examples, therefore we only need to align the configuration of these examples across their different plateform-specific implementations. In this PR, the configuration is as follows:
- On unix (x11, x99): `N` must be provided as a CLI argument to run the example. This is also used in our CI to run the test with a known number of messages. 
- On windows: since `getopt` is unavailable, `N` is initialized to `int32::max`.
- On freertos: I understand that CLI argument-passing may not be possible on certain embedded targets that run on freertox, therefore `N` is initialized to 10 to avoid running the example indefinitely. Should we change it to `int32::max` ?

Reply:
As per the other comment on this file, I'll update all implementations to start with a default number of samples of `int32:max`, and allow unix implementations to overwrite it through CLI argument.

Reply:
The behaviour of a `z_pub` in all languages is to publish indefinitely, so `uint32::MAX` is close enough :)